### PR TITLE
Adi/feedback submission api

### PR DESCRIPTION
**Pull request by AdiFinkelman**

### CODE REVIEW

File: src/main/kotlin/com/hibob/project/service/feedbackService.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class FeedbackService @Autowired constructor(private val feedbackDao: FeedbackDao) {
+    fun getAllFeedbacks(companyId: Long): List<FeedbackConfiguration> {
+        val feedbacks = feedbackDao.getAllFeedbacks(companyId)
+
+        if (feedbacks.isEmpty())
+            throw NotFoundException("There are no feedbacks for this company")
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/service/feedbackService.kt:14]: should this be an exception if my company doesn't have any feedback's yet?
Think about the user's exprience in this case

File: src/main/kotlin/com/hibob/project/service/feedbackService.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+        return feedbacks
+    }
+
+    fun feedbackSubmission(feedbackCreationRequest: FeedbackCreationRequest, employee: Employee) {
+        if (feedbackCreationRequest.text.length < 2)
+            throw IllegalArgumentException("Title must be at least 2 characters")
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/service/feedbackService.kt:21]: how will this function look if we have more validations?

Comment (vasofi) [reply src/main/kotlin/com/hibob/project/service/feedbackService.kt:21]: In this case the function is very short, and it's redundant, but something to think about

File: src/test/kotlin/com/hibob/project/service/FeedbackServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+class FeedbackServiceTest {
+    private val feedbackDao = mock<FeedbackDao>()
+    private val feedbackService = FeedbackService(feedbackDao)
+    val companyIdTest = 1234L
+
+    @Test
+    fun `get all feedbacks where feedback list is empty and throws exception`() {
+        val expectedMessage = assertThrows<NotFoundException> { feedbackService.getAllFeedbacks(companyIdTest) }
+        assertEquals("There are no feedbacks for this company", expectedMessage.message)
```
Comment (vasofi) [inline src/test/kotlin/com/hibob/project/service/FeedbackServiceTest.kt:23]: you don't really need to validate the excpetion message, this make the code highly coupled with the tests

Comment (AdiFinkelman) [reply src/test/kotlin/com/hibob/project/service/FeedbackServiceTest.kt:23]: We have learned to validate the exception message this way

File: src/main/kotlin/com/hibob/project/service/feedbackService.kt (hunk 2 above)
Comment (AdiFinkelman) [reply src/main/kotlin/com/hibob/project/service/feedbackService.kt:21]: I will expand it

File: src/main/kotlin/com/hibob/project/service/feedbackService.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+class FeedbackService @Autowired constructor(private val feedbackDao: FeedbackDao) {
+    fun getAllFeedbacks(companyId: Long): List<FeedbackConfiguration> {
+        val feedbacks = feedbackDao.getAllFeedbacks(companyId)
+        return feedbacks
+    }
+
+    fun feedbackSubmission(feedbackCreationRequest: FeedbackCreationRequest, employee: Employee) {
+        if (feedbackCreationRequest.text.length < 2)
+            throw IllegalArgumentException("Text must be at least 2 characters")
+
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/service/feedbackService.kt:17]: no blank line between if and else if, it should be clear that it is the same code block

Comment (vasofi) [reply src/main/kotlin/com/hibob/project/service/feedbackService.kt:17]: think of the function structure, it's called submit but it does validation and submit. Most of the code here is validation and if you need to add more validations, the code will become longer.
So think of the function structure here

**Review by vasofi — APPROVED** [review]

**Pull request merged**